examples/experimental/potree/src/pointcloud-tileset.ts

```diff
@@ -0,0 +1,72 @@
+import {Vector3} from '@math.gl/core';
+import {DataSource} from '@loaders.gl/loader-utils';
+
+/** Deck.gl Viewport instance type.
+ * We can't import it from Deck.gl to avoid circular reference */
+export type Viewport = {
+  id: string;
+  cameraPosition: number[] | Vector3;
+  height: number;
+  width: number;
+  zoom: number;
+  distanceScales: {
+    unitsPerMeter: number[];
+    metersPerUnit: number[];
+  };
+  center: number[] | Vector3;
+  unprojectPosition: (position: number[] | Vector3) => [number, number, number];
+  project: (coorinates: number[] | Vector3) => number[];
+};
+
+type PointcloudTilesetProps = {
+    /** Delay time before the tileset traversal. It prevents traversal requests spam.*/
+    debounceTime: number;
+  };
+  
+  const DEFAULT_PROPS: PointcloudTilesetProps = {
+    debounceTime: 0,
+  };
+
+export class PointcloudTileset {
+  private frameNumber: number = 0;
+  private lastUpdatedVieports: Viewport[] | Viewport | null = null;
+  private updatePromise: Promise<number> | null = null;
+
+  public options: PointcloudTilesetProps;
+
+  constructor(public dataSource: DataSource, options?: Partial<PointcloudTilesetProps>) {
+    // PUBLIC MEMBERS
+    this.options = {...DEFAULT_PROPS, ...options};
+  }
+
+  get isReady() {
+    // @ts-expect-error isReady property is not present in the abstract class
+    return this.dataSource.isReady;
+  }
+
+  get tiles() {
+    return [];
+  }
+
+  get isLoaded() {
+    return true;
+  }
+
+  async selectTiles(viewports: Viewport[] | Viewport | null = null): Promise<number> {
+    // @ts-expect-error init property is not present in the abstract class
+    await this.dataSource.init();
```

Comment:
Q: 
- we do have dataSource.isReady, does that not work?
- Do we need to await init? I don't see us referencing the dataSource in here, and I assume the dataset's methods are async anyway and internally await initialization?

Reply:
* isReady is a sync function. We have to await when metadata is loaded;
* init was already called under the hood. As it is called, it will return the promise of the initial call. Probably, this approach is not clear